source/main/cardano/CardanoNode.js

```diff
@@ -275,48 +275,61 @@ export class CardanoNode {
     );
 
     return new Promise(async (resolve, reject) => {
-      const logFile = createWriteStream(config.logFilePath, { flags: 'a' });
-      logFile.on('open', async () => {
-        this._cardanoLogFile = logFile;
-        // Spawning cardano-node
-        _log.info('CardanoNode path with args', {
-          path: walletBin,
-          args: walletArgs,
-        });
-
-        const node = await CardanoWalletLauncher({
-          path: walletBin,
-          nodeBin,
-          walletArgs,
-          logStream: logFile,
-          nodeImplementation,
-          cliBin,
-          stateDir: config.workingDir,
-        });
+      const logFile = rfs(
+        time => {
+          // The module works by writing to the one file name before it is rotated out.
+          if (!time) {
+            return `node.log-current`;
+          }
 
-        this._node = node;
-
-        try {
-          await promisedCondition(() => node.connected, startupTimeout);
-          // Setup livecycle event handlers
-          node.on('message', this._handleCardanoNodeMessage);
-          node.on('exit', this._handleCardanoNodeExit);
-          node.on('error', this._handleCardanoNodeError);
-          // Request cardano-node to reply with port
-          node.send({ QueryPort: [] });
-          _log.info(
-            `CardanoNode#start: cardano-node child process spawned with PID ${
-              node.pid
-            }`,
-            { pid: node.pid }
-          );
-          resolve();
-        } catch (_) {
-          reject(
-            new Error('CardanoNode#start: Error while spawning cardano-node')
-          );
+          return `node.log-${Date.now()}`;
+        },
+        {
+          size: '5M',
+          path: config.logFilePath,
+          maxFiles: 10,
```

Comment:
I think we need to set the limit of maxFiles to 4. Right @nikolaglumac ?

Reply:
No this is actually fine - we need to pack only 4 latest logs into our logs-download. Can you please make sure it works like that @DeeJayElly?